### PR TITLE
Removed deprecated files

### PR DESCRIPTION
**Pull request by ITPini**
## Description

The weak and unworthy will perish from this domain! 
Only the strong and righteous shall survive!

## Changes

- Removed deprecated files
- Dependency added: __None__
- Removed unused packages 
- Cleaned some code 

## Related Issues

None.

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 
Branch: `dev` (486ab30edfdbec6ce8637222b5562e90103f0433)

## Notes for Reviewers

### CODE REVIEW

**Review by MortenMunk — APPROVED** [review]
I did not run this locally, but it seems fine to me, and the deleted files seem like they were not in use. Good job 👍 🥇

File: eml/api/api.js [hunk 1]
```diff
@@ -257,27 +196,14 @@ export const fetchCertificates = async (userId) => {
...
-// Call to backend to see if online
-export const checkBackendOnline = async () => {
-  let response;
-  try {
-    const res = await axios.get(url + '/api/utility/online/', {timeout: 1000});
-    response = res.data;
-  } catch {
-    response = false;
-  }
-  return response;
```
Comment (sofiiagran) [inline eml/api/api.js:270]: why delete?

File: eml/api/api.js [hunk 2]
```diff
@@ -295,26 +221,9 @@ export const getVideoStreamUrl = (fileName, resolution) => {
...
-  try {
-    const res = await axios.get(url + '/api/sections/' + sectionId);
-    return res.data;
-  } catch (err) {
-    if (err?.response?.data != null) {
-      throw err.response.data;
-    } else {
-      throw err;
-    }
-  }
```
Comment (sofiiagran) [inline eml/api/api.js:314]: why delete?

File: eml/api/api.js [hunk 3]
```diff
@@ -335,8 +244,7 @@ export const getBucketImage = async (fileName) => {
     const res = await axios.get(
       `${url}/api/bucket/${fileName}`
       , {timeout: timeoutInMs});
-    const workingUrl = `data:image/png;base64,${res.data}`;
```
Comment (sofiiagran) [inline eml/api/api.js:338]: is it right to delete this?

File: eml/components/courses/courseCard/CourseCard.jsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ export default function CourseCard({ course, isOnline }) {
 
   const layout = downloaded || isOnline ? enabledUI : disabledUI;
 
-  let isDisabled = layout === disabledUI ? true : false;
+  let isDisabled = layout === disabledUI;
```
Comment (sofiiagran) [inline eml/components/courses/courseCard/CourseCard.jsx:34]: why change?

File: eml/components/courses/courseCard/DownloadCourseButton.jsx [hunk 5]
```diff
@@ -15,8 +15,7 @@ const ANIMATION_STATES = {
 
 /**
  * DownloadCourseButton component displays a button that downloads a course
- * @param {object} course - The course the button is on
- * @param {boolean} disabled - Whether the button is disabled or not
+ * @param {object} props - Props containing the course the button is on
```
Comment (sofiiagran) [inline eml/components/courses/courseCard/DownloadCourseButton.jsx:18]: just to make sure this is correct?

File: eml/screens/lectures/CompSwipeScreen.jsx [hunk 6]
```diff
@@ -65,11 +65,7 @@ export default function LectureSwipeScreen({ route }) {
     swiperRef.current.scrollBy(1, true);
     setScrollEnabled(true);
 
-    if (index === combinedLecturesAndExercises.length - 1) {
-      return true;
-    } 
-
-    return false;
+    return index === combinedLecturesAndExercises.length - 1;
   };
```
Comment (sofiiagran) [inline eml/screens/lectures/CompSwipeScreen.jsx:69]: I am not sure about this. Why would you change it?

File: eml/services/utilityFunctions.js [hunk 7]
```diff
@@ -241,13 +241,7 @@ export function findIndexOfUncompletedComp(student, courseId, sectionId) {
...
 
-  const indexOfUncompletedComp = section.components.findIndex(component => !component.isComplete);
-
-  if (indexOfUncompletedComp === -1) {
-    console.log(`All components in section ${sectionId} of course ${courseId} are complete.`);
-  }
-
-  return indexOfUncompletedComp;
+  return section.components.findIndex(component => !component.isComplete);
 }
```
Comment (sofiiagran) [inline eml/services/utilityFunctions.js:245]: just to make sure that this should be changed.. have you checked that it works?

**Review by sofiiagran — CHANGES_REQUESTED** [review]
Good work guys. I have some comments I would like you to consider, and I have not tested locally yet, I will do it tomorrow.

File: eml/api/api.js [hunk 8]
```diff
@@ -295,26 +221,9 @@ export const getVideoStreamUrl = (fileName, resolution) => {
     resolutionPostfix = '_360x640';
   }
 
-  const _vidUrl = `${url}/api/bucket/stream/${fileName}${resolutionPostfix}.mp4`;
-  return _vidUrl;
```
Comment (sofiiagran) [inline eml/api/api.js:299]: why delete this?

File: eml/api/api.js (hunk 1 above)
Comment (Donhusum) [reply eml/api/api.js:270]: This functionality is now done by the NetworkService functions, and so, this call is not in use any more

File: eml/components/courses/courseCard/CourseCard.jsx (hunk 4 above)
Comment (Donhusum) [reply eml/components/courses/courseCard/CourseCard.jsx:34]: The last part war redundant. The statement said if true then true, if false then false, so now it just take the boolean from the equality and don't do the whole trinary statement that returned the same thing

File: eml/screens/lectures/CompSwipeScreen.jsx (hunk 6 above)
Comment (Donhusum) [reply eml/screens/lectures/CompSwipeScreen.jsx:69]: Like in this https://github.com/Educado-App/educado-mobile/pull/221#discussion_r1410319842, it is redundant to say if true then true, if false then false.

File: eml/services/utilityFunctions.js (hunk 7 above)
Comment (Donhusum) [reply eml/services/utilityFunctions.js:245]: We removed the console.log (line 246-248) and then it is again assigned to a variable before returning the variable. Now it just returns it, no need for an extra step with no purpose.

File: eml/api/api.js (hunk 8 above)
Comment (Donhusum) [reply eml/api/api.js:299]: Again assigning the value to a variable before returning it. Redundant step removed

File: eml/components/courses/courseCard/DownloadCourseButton.jsx (hunk 5 above)
Comment (Donhusum) [reply eml/components/courses/courseCard/DownloadCourseButton.jsx:18]: As far as I can see it is correct

**Review by PrometheusBork — APPROVED** [review]
Works locally on my device for me

File: eml/api/api.js (hunk 1 above)
Comment (ITPini) [reply eml/api/api.js:270]: TODO: Remember to delete /api/utility/online/ route from backend

**Review by sofiiagran — APPROVED** [review]
Great job you guys.

I approve, but see the comment about the DownloadCourseButton(props).

It is just a comment, and it will not block this PR, but it may be a good idea to change:)

**Review by MortenMunk — APPROVED** [review]
Quality of life changes, but they are respectable 👍 🥇

**Review by amaliedilling — APPROVED** [review]

**Pull request merged**